notify/impl.go

```diff
@@ -678,12 +671,21 @@ func (n *Slack) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 
 	var numFields = len(n.conf.Fields)
 	if numFields > 0 {
-		var fields = make([]slackAttachmentField, numFields)
-		for k, v := range n.conf.Fields {
-			fields[k] = slackAttachmentField{
-				tmplText(v["title"]),
-				tmplText(v["value"]),
-				n.conf.ShortFields,
+		var fields = make([]config.SlackField, numFields)
+		for index, field := range n.conf.Fields {
+			// Check if short was defined for the field otherwise fallback to the global setting
+			var short bool
+			if field.Short {
```

Comment:
This won't work as expected when `field.Short` is set to `false` on purpose. `Short` needs to be a `*bool` instead.